client/header/activity-panel/panels/inbox/index.js

```diff
@@ -116,27 +126,28 @@ class InboxPanel extends Component {
 			);
 		}
 
+		const hasNotes = hasValidNotes( notes );
+
 		return (
 			<Fragment>
-				<ActivityHeader
-					title={ __( 'Inbox', 'woocommerce-admin' ) }
-					subtitle={ __(
-						'Insights and growth tips for your business',
-						'woocommerce-admin'
-					) }
-					unreadMessages={ this.getUnreadNotesCount() }
-				/>
+				{ ( hasNotes || isRequesting || isUndoRequesting ) && (
+					<ActivityHeader
+						title={ __( 'Inbox', 'woocommerce-admin' ) }
+						subtitle={ __(
+							'Insights and growth tips for your business',
+							'woocommerce-admin'
+						) }
+						unreadMessages={ getUnreadNotesCount(
+							notes,
+							lastRead
+						) }
+					/>
+				) }
+				{ this.renderNotePlaceholder( isRequesting, isUndoRequesting ) }
 				<Section>
-					{ isRequesting ? (
-						<ActivityCardPlaceholder
-							className="woocommerce-inbox-activity-card"
-							hasAction
-							hasDate
-							lines={ 2 }
-						/>
-					) : (
-						this.renderNotes()
-					) }
+					{ ! isRequesting &&
+						isUndoRequesting !== 'undo-dismiss-all' &&
+						this.renderNotes( hasNotes, isUndoRequesting ) }
```

Comment:
No need to pass in `isUndoRequesting` here.

Reply:
Nice catch! Solved in the commit  [8b64bb9](https://github.com/woocommerce/woocommerce-admin/pull/4379/commits/8b64bb904ab47025dc75cda8040e01eb8c21156d)